modules/services/sleepwatcher/default.nix

```diff
@@ -0,0 +1,148 @@
+        example = "echo 'Waking from sleep'";
+        description = "Commands to execute when system wakes up.";
+      };
+
+      on_display_sleep = mkOption {
+        type = types.nullOr types.lines;
+        default = null;
+        example = "echo 'Display sleeping'";
+        description = "Commands to execute when display goes to sleep.";
+      };
+
+      on_display_wakeup = mkOption {
+        type = types.nullOr types.lines;
+        default = null;
+        example = "echo 'Display awake'";
+        description = "Commands to execute when display wakes up.";
+      };
+
+      on_display_dim = mkOption {
+        type = types.nullOr types.lines;
+        default = null;
+        example = "echo 'Display dimmed'";
+        description = "Commands to execute when display dims.";
+      };
+
+      on_display_undim = mkOption {
+        type = types.nullOr types.lines;
+        default = null;
+        example = "echo 'Display undimmed'";
+        description = "Commands to execute when display undims.";
+      };
+
+      on_plug = mkOption {
+        type = types.nullOr types.lines;
+        default = null;
+        example = "echo 'Power connected'";
+        description = "Commands to execute when power adapter is plugged in.";
+      };
+
+      on_unplug = mkOption {
+        type = types.nullOr types.lines;
+        default = null;
+        example = "echo 'Power disconnected'";
+        description = "Commands to execute when power adapter is unplugged.";
+      };
+
+      idle_time = mkOption {
+        type = types.nullOr types.int;
+        default = null;
+        example = 600;
+        description = "Idle time in seconds before executing on_idle commands.";
+      };
+
+      on_idle = mkOption {
+        type = types.nullOr types.lines;
+        default = null;
+        example = "echo 'User idle'";
+        description = "Commands to execute when user becomes idle (requires idle_time to be set).";
+      };
+
+      on_idle_resume = mkOption {
+        type = types.nullOr types.lines;
+        default = null;
+        example = "echo 'User active again'";
+        description = "Commands to execute when user resumes activity after idle (requires idle_time to be set).";
+      };
+
+      run_as_system = mkOption {
+        type = types.bool;
+        default = false;
+        description = ''
+          Whether to run sleepwatcher as a system daemon (true) or user agent (false).
+          System daemon runs as root and can execute system-level commands.
+          User agent runs as the user and is preferred for user-specific actions.
+        '';
+      };
+    };
+  };
+
+  config = mkIf cfg.enable (
+    let
+      optionalHook = longArg: hookValue:
+        let hookScript = pkgs.writeShellScript "sleepwatcher-${longArg}-hook" hookValue;
+        in optionals (hookValue != null) [ "--${longArg}" "${hookScript}" ];
+
+      launchdConfig = {
+        path = [ config.environment.systemPath ];
```

Comment:
What's the reasoning behind this as opposed to having people set the PATH individually on each script that needs it set?